src/main/webapp/app/overview/course-statistics/course-statistics.component.ts

```diff
@@ -108,61 +106,49 @@ export class CourseStatisticsComponent implements OnInit, OnDestroy {
-        {
-            data: [0, 0, 0, 0, 0, 0],
-            backgroundColor: this.doughnutChartColors,
-        },
-    ];
-
-    public barChartOptions: any = {
-        scaleShowVerticalLines: false,
-        maintainAspectRatio: false,
-        responsive: true,
-        scales: {
-            xAxes: [
-                {
-                    stacked: true,
-                    ticks: {
-                        autoSkip: false,
-                        maxRotation: 0,
-                        minRotation: 0,
-                    },
-                    gridLines: {
-                        display: false,
-                    },
-                },
-            ],
-            yAxes: [
-                {
-                    stacked: true,
-                },
-            ],
-        },
-        tooltips: {
-            backgroundColor: 'rgba(0, 0, 0, 1)',
-            width: 120,
-            callbacks: {
-                label: (tooltipItem: any, data: any) => {
-                    return data.datasets[tooltipItem.datasetIndex].tooltips[tooltipItem.index];
-                },
-                afterLabel: (tooltipItem: any, data: any) => {
-                    return data.datasets[tooltipItem.datasetIndex].footer[tooltipItem.index];
-                },
-            },
-        },
-    };
-    public barChartType: ChartType = 'horizontalBar';
-
-    public doughnutChartType: ChartType = 'doughnut';
-    public totalScoreOptions: object = {
-        cutoutPercentage: 75,
-        scaleShowVerticalLines: false,
-        responsive: false,
-        tooltips: {
-            backgroundColor: 'rgba(0, 0, 0, 1)',
-        },
-    };
+    // ngx-charts
+    ngxDoughnutData: any[] = [];
+
+    // Labels for the different parts in Your overall points chart
+    programmingPointLabel = 'programmingPointLabel';
+    quizPointLabel = 'quizPointLabel';
+    modelingPointLabel = 'modelingPointLabel';
+    textPointLabel = 'textPointLabel';
+    fileUploadPointLabel = 'fileUploadPointLabel';
+    missingPointsLabel = 'missingPointsLabel';
+
+    ngxDoughnutColor = {
+        name: 'Your overall points color',
+        selectable: true,
+        group: ScaleType.Ordinal,
+        domain: this.doughnutChartColors, // colors: orange, turquoise, violet, bordeaux, green, red
+    } as Color;
+
+    // arrays representing each exercise group
+    ngxModelingExercises: any[] = [];
+    ngxProgrammingExercises: any[] = [];
+    ngxQuizExercises: any[] = [];
+    ngxFileUploadExercises: any[] = [];
+    ngxTextExercises: any[] = [];
+
+    // flags determining for each exercise group if at least one exercise has presentation score enabled
+    quizPresentationScoreEnabled = false;
+    programmingPresentationScoreEnabled = false;
+    modelingPresentationScoreEnabled = false;
+    textPresentationScoreEnabled = false;
+    fileUploadPresentationScoreEnabled = false;
+
+    ngxBarColor = {
+        name: 'Score per exercise group',
```

Comment:
Is this shown somewhere to a user? We should watch out for i18n then; if not, is this string re-used somewhere else? Should we encode these values in an enumeration?

Reply:
See my comments above to the color names